osu.Game/Rulesets/Difficulty/DifficultyCalculator.cs

```diff
@@ -179,7 +180,8 @@ static IEnumerable<Mod> createDifficultyAdjustmentModCombinations(ReadOnlyMemory
         /// Creates the <see cref="Skill"/>s to calculate the difficulty of an <see cref="IBeatmap"/>.
         /// </summary>
         /// <param name="beatmap">The <see cref="IBeatmap"/> whose difficulty will be calculated.</param>
+        /// <param name="visualMods">Visual mods to calculate difficulty with.</param>
         /// <returns>The <see cref="Skill"/>s.</returns>
-        protected abstract Skill[] CreateSkills(IBeatmap beatmap);
+        protected abstract Skill[] CreateSkills(IBeatmap beatmap, List<ModWithVisibilityAdjustment> visualMods);
```

Comment:
As I said already, I find constraining this to just "visual" mods sort of arbitrary. I think if we want skills to be aware of active mods, they should just receive *all* active mods - that will surely make even more things possible than passing just "visual" mods?

Reply:
Can agree with this. If we're going to make things more flexible there's no reason to have an arbitrary limit like this.

Reply:
I have no issue with simply giving skills access to all mods, however I don't see any need to and think it might encourage misuse of them. There is a fundamental difference with visual mods in that they are not reflected in the hitobject's values. Regular mods make physical transformations to the hitobjects which are used for difficulty calculation.

Reply:
If we're going to go in that direction, I'd have to add the argument that visual mods don't actually affect difficulty and probably shouldn't be considered, but that's a whole different discussion to be had.

Reply:
That is true in what we have always considered for the official gamemodes, but is it right for us to impose that limitation on custom ruleset developers?

Reply:
If we're looking for any kind of consistency.

At the point mods are contributing to difficulty a lot of considerations need to change. I'm not in the position to make a call like this, but I'd appreciate it not being hard-coded into the difficulty algorithm in the way you've done. It's either all or no mods.

Reply:
Changed in 95dcc73c742d4b02db88684656c57bf3916bd1f3.
Now passing all mods :+1: